include/envoy/network/connection.h

```diff
@@ -325,6 +328,19 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    */
   virtual absl::string_view transportFailureReason() const PURE;
 
+  /**
+   * @return std::string type of the underlying transport socket.
+   */
+  virtual std::string transportProtocol() const PURE;
```

Comment:
The only meaningful use of this one is to check whether transport socket can startSecureTransport, but I don't think you really need as you can call `startSecureTransport` and check its return value?

Reply:
Yeah if you want to reject at configuration level this should be part of transport socket factory but not connection.